Lib/test/test_code.py

```diff
@@ -141,6 +147,49 @@ def test_newempty(self):
         self.assertEqual(co.co_name, "funcname")
         self.assertEqual(co.co_firstlineno, 15)
 
+    @cpython_only
+    def test_closure_injection(self):
+        # From https://bugs.python.org/issue32176
+        from types import FunctionType, CodeType
+
+        def create_closure(__class__):
+            return (lambda: __class__).__closure__
+
+        def new_code(c_or_f):
+            '''A new code object with a __class__ cell added to freevars'''
+            c = c_or_f.__code__ if isinstance(c_or_f, FunctionType) else c_or_f
+            return CodeType(
+                c.co_argcount, c.co_kwonlyargcount, c.co_nlocals,
+                c.co_stacksize, c.co_flags, c.co_code, c.co_consts, c.co_names,
+                c.co_varnames, c.co_filename, c.co_name, c.co_firstlineno,
+                c.co_lnotab, c.co_freevars + ('__class__',), c.co_cellvars)
+
+        def add_foreign_method(cls, f):
+            code = new_code(f.__code__)
+            name = f.__name__
+            defaults = f.__defaults__
+            closure = (f.__closure__ or ()) + create_closure(cls)
+            setattr(cls, name, FunctionType(code, globals(), name, defaults, closure))
+
+        class List(list):
+            def append(self, elem):
+                super().append(elem)
+            def extend(self, elems):
+                super().extend(elems)
+
+        add_foreign_method(List, __getitem__)
+
+        # Ensure the closure injection actually worked
+        function = List.__getitem__
+        class_ref = function.__closure__[0].cell_contents
+        self.assertIs(class_ref, List)
+
+        # Ensure the code correctly indicates it accesses a free variable
+        self.assertFalse(function.__code__.co_flags & inspect.CO_NOFREE)
+
+        # Ensure the implicit super() call actually works
+        obj = List([1,2,3])
+        self.assertEqual(obj[0], 1)
```

Comment:
Does it print to stdout? stdout shouldn't be polluted by tests in non-verbose mode.

How do you know that the injected method actually works instead of the original list method? I suggest to transform the value in `__getitem__`. For example by calling `str()`.

Reply:
Done - replaced the print() with the use of an f-string as the getitem result.